src/emoji/emojiSelectors.js

```diff
@@ -1,13 +1,14 @@
 /* @flow strict-local */
 import { createSelector } from 'reselect';
 import type { Selector, RealmEmojiById, ImageEmojiType } from '../types';
+import { assumeSecretlyGlobalState } from '../reduxTypes';
 import { getRawRealmEmoji } from '../directSelectors';
 import { getIdentity } from '../account/accountsSelectors';
 import zulipExtraEmojiMap from './zulipExtraEmojiMap';
 import { objectFromEntries } from '../jsBackport';
 
 export const getAllImageEmojiById: Selector<RealmEmojiById> = createSelector(
-  getIdentity,
+  state => getIdentity(assumeSecretlyGlobalState(state)),
```

Comment:
```
redux types: Add two assumeSecretlyGlobalState markings
```

Just to check my understanding: Is it intentional that `getIdentity` is still a `Selector` at the time this change is made, and the switch to `GlobalSelector` comes after?

In the commit message, I see "[…] invokes one of those selectors (which demands a GlobalState)", and I think I'm a little confused because at this commit `getIdentity` doesn't yet demand a `GlobalState`, by that name, anyway.

Reply:
Yeah, I guess that is kind of talking about the future. Probably it'd be clearer to switch things to GlobalSelector where appropriate, then add these two markings, then switch other selectors to PerAccountState where appropriate (and then make the change that distinguishes PerAccountState from GlobalState.) I'll quick make that change.